core/src/main/java/novoda/lib/sqliteprovider/util/DatabaseStructure.java

```diff
@@ -0,0 +1,226 @@
+package novoda.lib.sqliteprovider.util;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import novoda.lib.sqliteprovider.sqlite.IDatabaseMetaInfo.SQLiteType;
+
+public class DatabaseStructure {
+
+    private static final String SELECT_TABLES_NAME = "SELECT name FROM sqlite_master WHERE type='table';";
+    private static final String PRAGMA_TABLE_INFO = "PRAGMA table_info('%1$s');";
+    private static final String PRGAMA_INDEX_LIST = "PRAGMA index_list('%1$s');";
+    private static final String PRGAMA_INDEX_INFO = "PRAGMA index_info('%1$s');";
+    private static final String COLUMN_PRIMARY_KEY_INDEX = "pk";
+    private static final String COLUMN_NAME = "name";
+    private static final String COLUMN_TYPE = "type";
+
+    private static final List<String> DEFAULT_TABLES = Collections.singletonList("android_metadata");
+
+    private final SQLiteDatabase database;
+
+    private List<String> createdTables;
```

Comment:
I slightly agree on the error-proneness of these fields, but I decided on having them so I can avoid using output parameters (F2). Also, you may call YAGNI on this, but it opens possibilities for caching.